proof/invariant-abstract/DetSchedDomainTime_AI.thy

```diff
@@ -117,27 +86,12 @@ crunches update_restart_pc
 locale DetSchedDomainTime_AI_2 = DetSchedDomainTime_AI +
   assumes handle_hypervisor_fault_domain_list_inv'[wp]:
     "\<And>P t f. \<lbrace>\<lambda>s::det_state. P (domain_list s)\<rbrace> handle_hypervisor_fault t f \<lbrace>\<lambda>_ s. P (domain_list s)\<rbrace>"
-  assumes handle_hypervisor_fault_domain_time_inv'[wp]:
-    "\<And>P t f. \<lbrace>\<lambda>s::det_state. P (domain_time s)\<rbrace> handle_hypervisor_fault t f \<lbrace>\<lambda>_ s. P (domain_time s)\<rbrace>"
   assumes arch_perform_invocation_domain_list_inv'[wp]:
     "\<And>P i. \<lbrace>\<lambda>s::det_state. P (domain_list s)\<rbrace> arch_perform_invocation i \<lbrace>\<lambda>_ s. P (domain_list s)\<rbrace>"
-  assumes arch_perform_invocation_domain_time_inv'[wp]:
-    "\<And>P i. \<lbrace>\<lambda>s::det_state. P (domain_time s)\<rbrace> arch_perform_invocation i \<lbrace>\<lambda>_ s. P (domain_time s)\<rbrace>"
-  assumes handle_interrupt_valid_domain_time:
-    "\<And>i.
-      \<lbrace>\<lambda>s :: det_state. 0 < domain_time s \<rbrace>
-        handle_interrupt i
-      \<lbrace>\<lambda>rv s.  domain_time s = 0 \<longrightarrow> scheduler_action s = choose_new_thread \<rbrace>"
-  assumes handle_reserved_irq_some_time_inv'[wp]:
-    "\<And>P irq. \<lbrace>\<lambda>s::det_state. P (domain_time s)\<rbrace> handle_reserved_irq irq \<lbrace>\<lambda>_ s. P (domain_time s)\<rbrace>"
   assumes handle_reserved_irq_domain_list_inv'[wp]:
     "\<And>P irq. \<lbrace>\<lambda>s::det_state. P (domain_list s)\<rbrace> handle_reserved_irq irq \<lbrace>\<lambda>_ s. P (domain_list s)\<rbrace>"
   assumes arch_mask_irq_signal_domain_list_inv'[wp]:
     "\<And>P irq. arch_mask_irq_signal irq \<lbrace>\<lambda>s::det_state. P (domain_list s)\<rbrace>"
-  assumes arch_mask_irq_signal_domain_time_inv'[wp]:
-    "\<And>P irq. arch_mask_irq_signal irq \<lbrace>\<lambda>s::det_state. P (domain_time s)\<rbrace>"
```

Comment:
Very nice to see so many assumptions of this locale removed (and so many lemmas in this file deleted)!